mobile/android/android-application-manager.ts

```diff
@@ -49,10 +50,39 @@ export class AndroidApplicationManager extends ApplicationManagerBase {
 			"1"]);
 
 		if (!this.$options.justlaunch) {
+			let processIdentifier = await this.getProcessId(appIdentifier);
+			if (processIdentifier) {
+				let deviceIdentifier = this.adb.identifier;
+				this.$deviceLogProvider.setApplicationPidForDevice(deviceIdentifier, processIdentifier);
+			}
+
 			await this.$logcatHelper.start(this.identifier);
 		}
 	}
 
+	private async getProcessId(appId: string): Promise<string> {
+		// Read the process id of the application that was just started
+		// Information is returned in the following format
+		// USER      PID   PPID  VSIZE  RSS   WCHAN            PC  NAME
+		// u0_a64    4831  488   851380 94888    ep_poll f1e43bb9 S org.nativescript.myApplication
+		let processes = await this.adb.executeShellCommand(["ps"]);
+		processes = processes.trim()
+			.split("\n")
+			.map((line: string) => line.replace(/\s+/g, ' ')
+				.split(" ")
+				.filter((entry: string) => /\S/.test(entry)));
+
+		let currentProcessLine = processes.find((line: Array<string>) => line[line.length - 1] === appId);
+
+		if (!currentProcessLine) {
+			return Promise.resolve(null);
```

Comment:
you do not need to return Promise here, `async` keyword does this for you. 
```TypeScript
return null;
```
is enough